### PR TITLE
Replace overview SLO buttons with a six-stop slider / 将 Overview 页 SLO 按钮替换为六档滑块

### PR DESCRIPTION
**Pull request by functionstackx**
## Summary

- Replaces the six SLO buttons on `/overview` with one discrete range slider for 30, 50, 75, 100, 150, and 200 tok/s/user.
- Preserves the existing cached JSON navigation, canonical URL query, pending filter state, and `overview_selector_changed` analytics.
- Adds a selected-track treatment and contrasting ridges at all six stops, plus accessible English/Chinese value text, while retaining native keyboard, pointer, and touch behavior.
- Updates the overview E2E suite to exercise slider selection, ridge states, cache reuse, filter preservation, responsive layout, and the `/zh/overview` sibling.

## Validation

- `bun run fmt`
- `bun run lint`
- `bun run typecheck`
- Focused overview unit tests: 102 passed
- Fixture-backed `overview.cy.ts`: 28 passed
- Live browser verification at `/overview` and `/zh/overview` for SLO 50 and 100, including filled/selected/unfilled ridges; no console errors
- `bun run test:unit`: 3,705 app/constants/database tests passed; the unrelated MCP package then failed 25 tests during mocked server setup because `z.enum` was undefined

## 中文说明

- 将 `/overview` 原有的六个 SLO 按钮替换为离散滑块，可选择 30、50、75、100、150、200 tok/s/user 六档。
- 保留现有的 JSON 缓存导航、规范化 URL 查询参数、待提交筛选状态与 `overview_selector_changed` 埋点。
- 增加已选区间样式及六个档位的对比刻度脊线，并提供中英文无障碍数值说明，同时继续使用原生键盘、指针与触控交互。
- 更新 Overview E2E 测试，覆盖滑块选择、刻度状态、缓存复用、筛选条件保留、响应式布局及 `/zh/overview` 中文页面。

## 验证结果

- `bun run fmt`
- `bun run lint`
- `bun run typecheck`
- Overview 相关单元测试：102 项通过
- 基于 fixture 的 `overview.cy.ts`：28 项通过
- 在 `/overview` 与 `/zh/overview` 上对 SLO 50、100 完成浏览器实测（包括已填充、当前选中与未填充刻度），控制台无报错
- `bun run test:unit`：App、Constants、Database 共 3,705 项测试通过；随后与本次改动无关的 MCP package 在 mock server 初始化阶段因 `z.enum` 为 undefined 导致 25 项失败

---

> [!NOTE]
> **Medium Risk**
> Changes the primary overview filter control and its navigation edge cases (drag, errors, history), though behavior is heavily covered by E2E and unit tests.
> 
> **Overview**
> Replaces the **six discrete SLO tier links** on `/overview` with a **single six-stop range slider** (30–200 tok/s/user). `OverviewTierSwitcher` now renders the new `OverviewTierSlider` instead of anchor-based options—tier changes go through the same client `navigation.push` + `overviewTierHref` flow, `overview_selector_changed` analytics, and `?tier=` URL semantics as before.
> 
> The slider adds **visual ridges**, a filled track, and **`aria-valuetext`** for accessibility; commits on **pointer release** (drag) or **keyboard change**, with guards so mid-drag API updates do not move the thumb and **failed loads can be retried** without extra history entries. Tier is no longer chosen via copyable per-tier links inside the SLO control (other switchers still use hrefs).
> 
> **Tests** add a Cypress `selectOverviewTier` helper (hydration-safe range input), retarget assertions to `data-tier` / ridge states, extend the failed-request retry case, and move some unit/focus scenarios to **engine scope** links where tier is no longer a link.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 410ad979ee60e1ac22c37edac45df6c969ca11d0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: packages/app/src/components/overview/overview-tier-slider.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    setSelectedIndex(index);
+  }, [tier]);
+
+  const selectedTier = OVERVIEW_TIERS[selectedIndex];
+  const progress = (selectedIndex / (OVERVIEW_TIERS.length - 1)) * 100;
+
+  const commit = (index: number) => {
+    const nextTier = OVERVIEW_TIERS[index];
+    if (index === committedIndex.current) return;
+    committedIndex.current = index;
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/overview/overview-tier-slider.tsx:76]: Fixed in 410ad979: when navigation errors, an effect resets `committedIndex` from the settled tier, so activating the unsuccessful stop again re-runs `navigation.push` (which dedupes the byte-identical href, so no extra history entry). Covered by a new e2e assertion in the failure spec.

File: packages/app/src/components/overview/overview-tier-slider.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  const navigation = useOverviewNavigation();
+  const pointerActive = useRef(false);
+  const committedIndex = useRef(tierIndex(tier));
+  const [selectedIndex, setSelectedIndex] = useState(() => tierIndex(tier));
+
+  useEffect(() => {
+    const index = tierIndex(tier);
+    committedIndex.current = index;
+    setSelectedIndex(index);
+  }, [tier]);
```
[the remark this answers is not in the document]
Comment (functionstackx) [reply packages/app/src/components/overview/overview-tier-slider.tsx:61]: Fixed in 410ad979: the settled-tier effect still syncs `committedIndex` but skips `setSelectedIndex` while `pointerActive.current` is true, so a payload settling mid-drag no longer moves the thumb; the gesture's own pointerup/cancel settles it.
